### PR TITLE
Check for ENV vars that interfere with local testing

### PR DESCRIPTION
**Pull request by mfurtak**
Help us discover local state problems that interfere with releases

### Checklist
- [X] I've run `bundle exec rspec` from the root directory to see all new and existing tests pass
- [X] I've followed the _fastlane_ code style and run `bundle exec rubocop -a` to ensure the code style is valid
- [X] I've read the [Contribution Guidelines](https://github.com/fastlane/fastlane/blob/master/CONTRIBUTING.md)
- [X] I've updated the documentation if necessary.

### CODE REVIEW

**Comment by lacostej** [comment]
Could we unset them instead (and maybe warn the user?)

File: fastlane/Fastfile [hunk 1]
```diff
@@ -228,6 +228,18 @@ end
 
 desc "Verifies all tests pass and the current state of the repo is valid"
 private_lane :validate_repo do
+  # Ensure that no ENV vars which interefere with testing are set
```
Comment (KrauseFx) [inline fastlane/Fastfile]: Is the `Fastfile` the right spot to put this code? Shouldn't this be part of the test suite instead?

**Comment by mfurtak** [comment]
@lacostej Either is probably fine, but I think my preference is make our mistakes noisy, rather than ignorable.

File: fastlane/Fastfile (hunk 1 above)
Comment (mfurtak) [inline fastlane/Fastfile]: I suppose? I guess I look at my normal testing process to be more fluid. Maybe I'm setting ENV variables to explore things... In comparison, the release process should be pretty locked down.

Does that make sense? If not, do you have a sense of where you would put it?

Comment (KrauseFx) [inline fastlane/Fastfile]: I think `spec_helper` is the perfect place for this 👍

**Review by KrauseFx — APPROVED** [review]
A great update 👍

**Pull request merged**